app/assets/javascripts/pretty-text/engines/discourse-markdown/mentions.js.es6

```diff
@@ -20,14 +20,40 @@ function addMention(buffer, matches, state) {
 export function setup(helper) {
   helper.registerOptions((opts, siteSettings) => {
     opts.features.mentions = !!siteSettings.enable_mentions;
+    opts.features.unicodeUsernames = !!siteSettings.unicode_usernames;
   });
 
   helper.registerPlugin(md => {
     const rule = {
-      matcher: /@(\w[\w.-]{0,58}\w)|@(\w)/,
+      matcher: mentionRegex(md.options.discourse.features.unicodeUsernames),
       onMatch: addMention
     };
 
     md.core.textPostProcess.ruler.push("mentions", rule);
   });
 }
+
+function mentionRegex(unicodeUsernames) {
+  if (unicodeUsernames) {
+    try {
+      // Create the regex from a string, because Babel doesn't understand
```

Comment:
Would upgrading babel help here? Or is this always true?

Reply:
Maybe. I didn't test if a newer version of Babel uses a version of [babel-plugin-transform-unicode-regex](https://babeljs.io/docs/en/next/babel-plugin-transform-unicode-regex.html) that doesn't have that problem anymore. We certainly don't want to use [babel-plugin-proposal-unicode-property-regex](https://babeljs.io/docs/en/next/babel-plugin-proposal-unicode-property-regex.html) because transpiling Unicode property escapes doesn't create a perfect result. I've already added such a transpiled regex as fallback for browsers that do not support this feature yet.